crates/sui-json-rpc/src/name_service.rs

```diff
@@ -144,13 +150,17 @@ impl FromStr for Domain {
             ));
         }
 
-        let labels = s
-            .split('.')
+        let separator = separator(s)?;
+        let formatted_string = convert_from_new_format(s, &separator)?;
```

Comment:
At the end of this function, you join up the the vector of parts you have to create a single string, and the next step you perform below splits them again according to the separator -- why not leave it as a `Vec<String>` as the output of `convert_from_new_format`?